src/lib.rs

```diff
@@ -26,6 +26,9 @@ mod mqtt_client;
 #[cfg(feature = "mqtt-client")]
 pub use mqtt_client::*;
 
+#[cfg(feature = "mqtt-client")]
+pub use minimq;
```

Comment:
Do we really need this?
Or can we make the part of minimq that miniconf exposes so flexible that using it doesn't require access to additional minimq api to use it? Configbuilder is one of the main blockers here, right?

ref https://github.com/rust-lang/api-guidelines/discussions/176 for some discussion.

Reply:
The minimal API surface would be the `minimq::ConfigBuilder` and all the types that touch it (i.e. buffer config enum, `ProtocolError`, and `Broker` types).

That discussion seems to explicitly recommend a `pub use minimq` here, so I'm a bit confused

Reply:
I think I pasted the wrong link. I has read somewhere else (and that sounds very reasonable to me) that because of possible semver hazards, the re-exports should be absolutely minimal to allow meeting type requirements of the re-exporting crate but to minimize knock-on breaking changes due unrelated stuff changing in the re-exported crate.
Strictly we should only be re-exporting configbuilder, broker, and the relevant error type.
But that's already a lot, hence let's stick with a full re-export.